src/manage_nsfs/manage_nsfs_validations.js

```diff
@@ -738,6 +740,39 @@ function validate_connection_args(user_input, action) {
     }
 }
 
+////////////////////////////
+//// UPGRADE VALIDATION ////
+///////////////////////////
+
+/**
+ * validate_expected_version checks that expected_version is valid
+ * if it's not valid it throws an error
+ * @param {string} expected_version
+ */
+function validate_expected_version(expected_version) {
+    if (!expected_version) {
+        throw new Error('expected_version flag is required');
+    }
+    if (!version_utils.is_valid_sematic_version(expected_version)) {
+        throw new Error('expected_version must have sematic version structure (major.minor.patch)');
+    }
+    if (check_if_expected_version_newer_that_package_version(expected_version)) {
+        throw new Error(`expected_version cannot be later that the package version ${pkg.version}`);
+    }
+}
+
+/**
+ * compare_version_to_current_version checks that the version
+ * is not later that the one that was defined in the pkg
+ * @param {string} expected_version
+ * @returns {boolean}
+ */
+function check_if_expected_version_newer_that_package_version(expected_version) {
+    const package_version = pkg.version;
+    const diff_from_package_version = version_utils.version_compare(expected_version, package_version);
```

Comment:
in the bug, the error is about config_dir_version and not source code version
so that's a nice change but we probably should check how we behave on source code valid upgrade while in the config dir version there is nothing to upgrade because they match

Reply:
I added the error thrown in a separate commit.

Reply:
I think there is a misunderstanding here, 
when upgrading the config dir, the package.version must be equal to the expected_version 
we already throw on this issue in the first check of _verify_config_dir_upgrade(), and this verification can be disabled, this is something we should consider